doozer/doozerlib/cli/scan_osh.py

```diff
@@ -64,6 +68,60 @@ def __init__(self, runtime: Runtime, last_brew_event: int, dry_run: bool, check_
         # Initialize JIRA client
         self.jira_client: JIRA = self.runtime.build_jira_client()
 
+    # Static functions
+    @staticmethod
+    def is_jira_workflow_component_disabled(meta):
+        """
+        Check if the OCPBUGS workflow is enabled in ocp-build-data image meta config
+        """
+        flag = meta.config.external_scanners.sast_scanning.jira_integration.enabled
+
+        # Disable only if specifically defined in config
+        return flag in [False, "False", "false", "no"]
+
+    def is_jira_workflow_group_disabled(self):
+        """
+        Check if JIRA workflow is disabled for this particular OCP version in group.yml
+        """
+        flag = self.runtime.group_config.external_scanners.sast_scanning.jira_integration.enabled
+        return flag not in [True, "True", "true", "yes"]
+
+    @staticmethod
+    def get_substring_regex(pattern, string):
+        match = re.search(pattern=pattern, string=string)
+        if match:
+            return match.group("search_key")
```

Comment:
Why write a function for this? Also since we rely on search_key being in regex in the calling code,
it is better if we have both of them in the same place

Reply:
Well I needed to do string extractions using regex in more than one place, so I wrote the function to reduce redundancy. 

> Also since we rely on search_key being in regex in the calling code
Ahh yes, I missed this during refinement. Can I pass the search_key as a variable inside the function? Like

```
@staticmethod
    def get_substring_regex(pattern, string, search_key):
        match = re.search(pattern=pattern, string=string)
        if match:
            return match.group(search_key)
```

Reply:
like this https://github.com/openshift-eng/art-tools/pull/247/commits/4a74f2e74309158f496b9b17bad8a915442916b5

Reply:
Imo it's okay to have this kind of redundancy, it helps with readability and maintainability. Defining a function to wrap an if statement, that we later check with an assert statement isn't reducing the cognitive burden to understand what's going on.

Reply:
makes sense, will make the change

Reply:
updated in https://github.com/openshift-eng/art-tools/pull/247/commits/7a0cbcff199ae0e320344b186720e19dc0a292eb